app/models/profile.rb

```diff
@@ -35,7 +35,7 @@ class Profile < ApplicationRecord
 
   before_save do
     self.build_tags
-    self.construct_full_name
+    construct_full_name if full_name.nil?
```

Comment:
`full_name` in the database is currently used for search, and therefore always lowercase, so we should keep that behavior at the moment and ignore the `full_name` from the archive until we can handle `full_name` properly.